src/Application/Actions/Donations/Confirm.php

```diff
@@ -119,6 +119,7 @@ protected function action(Request $request, Response $response, array $args): Re
         // that, would like to find a way to make it so if its left out we get an error instead - either by having
         // derive fees return a value, or making functions like Donation::getCharityFeeGross throw if called before it.
         $this->donationRepository->deriveFees($donation, $cardBrand, $cardCountry);
+        $this->entityManager->persist($donation);
```

Comment:
AFAIK `persist` is only effective when called on a new entity that the ORM doesn't know about yet. Since $donation here came from the ORM in the first place I think persist does nothing:

https://www.doctrine-project.org/projects/doctrine-orm/en/2.16/reference/working-with-objects.html

> If X is a preexisting managed entity, it is ignored by the persist operation. However, the persist operation is cascaded to entities referenced by X, if the relationships from X to these other entities are mapped with cascade=PERSIST or cascade=ALL (see "[Working with Associations](https://www.doctrine-project.org/projects/doctrine-orm/en/2.16/reference/working-with-associations.html#transitive-persistence)").

The call to `flush` further down should be enough to save any changes to the donation, not sure why it hasn't been working.